src/Sentry/SentryEvent.cs

```diff
@@ -286,6 +286,24 @@ public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
     /// </summary>
     public static SentryEvent FromJson(JsonElement json) => FromJson(json, null);
 
+    private static SentryLevel? SafeLevelFromJson(JsonElement json)
+    {
+        var levelString = json.GetPropertyOrNull("level")?.GetString();
+        if (levelString == null)
+            return null;
+
+        // Native SentryLevel.None does not exist in dotnet
+        return levelString.ToLowerInvariant() switch
+        {
+            "debug" => SentryLevel.Debug,
+            "info" => SentryLevel.Info,
+            "warning" => SentryLevel.Warning,
+            "fatal" => SentryLevel.Fatal,
+            "error" => SentryLevel.Error,
+            _ => null
```

Comment:
I take it this is intentional (we don't want to throw an exception but we can't conveniently log the irregularity anywhere)?